v0/destinations/fb/transform.js

```diff
@@ -97,43 +115,75 @@ function sanityCheckPayloadForTypesAndModifications(updatedEvent) {
   updatedEvent.event = "CUSTOM_APP_EVENTS";
 }
 
+function getCorrectedTypedValue(pathToKey, value, originalPath) {
+  const type = eventPropToTypeMapping[pathToKey];
+  // TODO: we should remove this eslint rule or comeup with a better way
+  if (typeof value === type) {
+    return value;
+  }
+
+  throw new Error(
+    `${
+      typeof originalPath === "object"
+        ? JSON.stringify(originalPath)
+        : originalPath
+    } is not of valid type`
+  );
+}
+
 function processEventTypeGeneric(message, baseEvent, fbEventName) {
   const updatedEvent = {
     ...baseEvent
   };
   set(updatedEvent.custom_events[0], "_eventName", fbEventName);
 
-  Object.keys(message.properties).forEach(k => {
-    if (eventPropsToPathMapping[k]) {
-      let rudderEventPath = eventPropsToPathMapping[k];
-      let fbEventPath = eventPropsMapping[rudderEventPath];
-
-      if (rudderEventPath.indexOf("sub") > -1) {
-        const [prefixSlice, suffixSlice] = rudderEventPath.split(".sub.");
-        const parentArray = get(message, prefixSlice);
-        updatedEvent.custom_events[0][fbEventPath] = [];
-
-        let length = 0;
-        let count = parentArray.length;
-        while (count > 0) {
-          const intendValue = get(parentArray[length], suffixSlice);
-          updatedEvent.custom_events[0][fbEventPath][length] =
-            intendValue || "";
-
-          length++;
-          count--;
+  const { properties } = message;
+  if (properties) {
+    if (properties.revenue && !properties.currency) {
+      throw new Error(
+        "If properties.revenue is present, properties.currency is required."
```

Comment:
Can we set to something default like `USD`? Does FB do it?

Reply:
I think better to have explicit ?
https://developers.facebook.com/docs/marketing-api/app-event-api (fb_currency)